### DISCUSSION TITLE
MACOS Sierra clone problem

### DISCUSSION THREAD

**Discussion by akras123** [community/community#21539] [New to GitHub]
<p>Hi guys,</p>
<p>I’m a novice to Github and struggling with what might be very basic…</p>
<p>When I start a new repo in GH pages, then want to clone it to GH Desktop, I get an error message: Safari complains that it is unable to open a link that starts with:</p>
<p>‘x-github-client://openRepo/etc.’ Have reinstalled the Desktop, rebooted de compuer, same result.</p>
<p>The other way around: starting a simple repo with Desktop and Atom, with an index.md containing a link to a page2, it works fine on the local server, but when I upload it to GH pages, only the index file shows correctly, when clicking the link to the second page, safari does not find it. Safari shows the url of the index page correctly: <a class="inline-onebox-loading">~github.io/css2</a>. But for page2 it searches for <a class="inline-onebox-loading">~github.io/page2</a>, instead of <a class="inline-onebox-loading">~github.io/css2/page2</a>. Very confusing for me!</p>
<p> Did I screw up with a setting somewhere? I have no clue where to look, so help would be really appreciated!</p>

**Comment by username1290** [comment]
<p>You have a couple options, but first of all I suggest starting with:</p>
<p><strong>Create a repo in <a href="http://github.com" rel="nofollow noopener">github.com</a>, and check "add Readme"</strong></p>
<p>Now, the options:</p>
<p>A) Try the “open in desktop” using a different browser, i.e. <strong>Firefox</strong> or <strong>Chrome</strong></p>
<p>B) Using the <strong>GitHub desktop app</strong> :</p>
<p>1.- click the <strong>plus sign (+)</strong> at the Top Left</p>
<p><img src="https://user-images.githubusercontent.com/2784323/181065364-71dc6545-f90f-4425-9903-3f3b225c0e96.png" alt="Screen Shot 2017-11-14 at 12.08.19 AM.png" data-base62-sha1="1MHYuedI9XsPYQytQNU6Thmax6o" width="90" height="58"></p>
<p>2.- Select <strong>Clone</strong></p>
<p>3.- Find your repo, select it</p>
<p>4.- In the dialog window, select a folder in your computer (I recommend creating a folder named <strong>github</strong> inside Documents), click Ok</p>
<p>5.- Done! You’ll have the repo in your computer </p>
<p>Hope that helps :slight_smile:</p>
<p>6.- Now you can add files in this repo, and click <strong>Submit</strong>  and <strong>Sync</strong> to add them to GitHub</p>

**Comment by akras123** [comment]
<p>Thank you very much indeed, cloning from within the desktop works OK. Is this a known problem for Safari though, or did I do something wrong there?</p>
<p>As for my second question: I created a small repo with a mainmenu at the top. It looks like this: </p>
<p>&lt;link rel=“stylesheet” type=“text/css” href="/assets/css/mystyle.css"&gt;<br>
&lt;ul&gt;<br>
&lt;li&gt;&lt;a class=“active” href="/index/"&gt;Home&lt;/a&gt;&lt;/li&gt;<br>
&lt;li&gt;&lt;a href="/snippets/"&gt;Snippets&lt;/a&gt;&lt;/li&gt;<br>
&lt;li class=“dropdown”&gt;&lt;a href=“javascript:void(0)” class=“dropbtn”&gt;Dropdown&lt;/a&gt;<br>
&lt;div class=“dropdown-content”&gt;<br>
&lt;a href="/page3A/"&gt;Page 3A&lt;/a&gt;<br>
&lt;a href="/page3B/"&gt;Page 3B&lt;/a&gt;<br>
&lt;a href="/page3C/"&gt;Page 3C&lt;/a&gt;<br>
&lt;/div&gt;<br>
&lt;/li&gt;<br>
&lt;li&gt;&lt;a href="/about-us/"&gt;About&lt;/a&gt;&lt;/li&gt;<br>
&lt;/ul&gt;&lt;br/&gt;&lt;hr/&gt;</p>
<p>Works fine as long as it is on the local server, but when I sync it to Github Pages, all references seem to go wrong, resulting in “404 page not found”. The url Safari is looking for looks like: “<a href="https://akras123.github.io/page3A/" rel="nofollow noopener">https://akras123.github.io/page3A/</a>”, which should be: “<a href="https://akras123.github.io/css2/page3A/" rel="nofollow noopener">https://akras123.github.io/css2/page3A/</a>”, so the site reference “css2/” is missing. Should I include some prepend somewhere?</p>

**Comment by l2dy** [comment]
<p>You should use relative links (remove the leading slash, e.g. replace “/page3A/” with “page3A/”).</p>

**Comment by akras123** [comment]
<p>Wish it was that simple! When I use that, it doesn’t run even on the local server, which makes sense, because the files are in the root directory css2, while the menu is in css2/_includes/mainmenu.html. So, imho I need absolute addressing, and that works fine on the local server.</p>
<p>But when uploading this to Github Pages, I get confused. I find the index file as expected: <a href="https://akras123.github.io/css2/" rel="nofollow noopener">https://akras123.github.io/css2/</a>. The menu is also found, but not formatted (so /assets/css/mystyle.css is apparently not found). When I click on any of the menu entries (like Page3A), Safari doesn’t find the page at all (404 page not found). Checking the address bar of Safari, it appears that Safari is looking for “<a class="inline-onebox-loading">~.io/Page3A/</a>” instead of “<a class="inline-onebox-loading">~.io/css2/Page3A/</a>”. In other words: the site’s base url seems to be missing somehow… When I insert that in the address bar, the page is found.</p>
<p>And once again, as long as it is on the local server the absolute addressing works perfectly as expected! I hope the description is clear enough, and that you can help me out of the dark…</p>

**Comment by l2dy** [comment, marked as answer]
<p>Here is a quick and dirty solution: <a href="https://ricostacruz.com/til/relative-paths-in-jekyll" rel="nofollow noopener">https://ricostacruz.com/til/relative-paths-in-jekyll</a>. Hope it helps.</p>

**Comment by akras123** [comment]
<p>Thanks a lot, that solves the riddle indeed. I had just run into a comparable solution, i.e. to insert {{site.baseurl}} in any absolute address. Works OK as well, and it seems I don’t need to include any extra coding for it, apparently site.baseurl is already predefined somewhere…</p>
<p>Anyway, now I can proceed ans discover further, thanks again!</p>
